.github/workflows/hexpm-mirror.yml

```diff
@@ -0,0 +1,35 @@
+---
+name: hexpm-mirror
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  test_option_hexpm_mirror:
+    name: Test option hexpm-mirror
+    runs-on: ${{matrix.os}}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: ['ubuntu-22.04', 'windows-2019']
+        otp-version: ['24']
+        elixir-version: ['v1.12', '']
```

Comment:
Can we move `v1.12` to `v1.14`?
Also, why is there a `''` here?

Reply:
`''` is there to have no elixir. The logic is a little bit different when elixir is not installed: we don't cache Windows OTP and if there is no Elixir on Windows, there should be no cache.

This is why I had to modify the function that gets the Elixir version, something that will be explained in a comment as you requested [here](https://github.com/erlef/setup-beam/pull/156#discussion_r1026763634).

Reply:
But I'm assuming we don't want to allow for a non-version value. In this case, you're passing in an empty string instead of `undefined`, right? Unless the code I asked to be commented is also executed in the `undefined` case. I'll wait for your comment.

Reply:
Passing an empty string here means that ${{ matrix.elixir-version }} will be replaced by an empty string, and it will be like [what happens with ubuntu.yml CI file](https://github.com/erlef/setup-beam/blob/f19acc5287cd038f617d35af560c4f3720732ad1/.github/workflows/ubuntu.yml#L144) for cases like the [first one](https://github.com/erlef/setup-beam/blob/f19acc5287cd038f617d35af560c4f3720732ad1/.github/workflows/ubuntu.yml#L24).

Reply:
Ok, I'll wait for the comment still 👍 since I don't see why we didn't need check that before but we do now. I do have to say I'd rather have undefined or a version, than allowing for empty string, but I am aware of how strange YAML and JavaScript are in relation to empty strings, undefined and the like.

Reply:
See also https://github.com/actions/toolkit/blob/819157bf872a49cfcc085190da73894e7091c83c/packages/core/src/core.ts#L120

Reply:
Ok, the good reason to justify this change was that the code was previously protected at https://github.com/erlef/setup-beam/pull/156/files#diff-6406ae11684199b17d2cdca1efe2cfa29fb4587c3b142f1d33ad5f81a5acf7adL60, and now `getElixirVersion` was moved out of that function.